File: api/src/main/java/io/github/kdy05/physicalFighters/api/PotionEffectTypeAdapter.java
package io.github.kdy05.physicalFighters.api;

import org.bukkit.potion.PotionEffectType;

public interface PotionEffectTypeAdapter {

    PotionEffectType SLOWNESS();
    PotionEffectType HASTE();
    PotionEffectType MINING_FATIGUE();
    PotionEffectType STRENGTH();
    PotionEffectType INSTANT_HEALTH();
    PotionEffectType INSTANT_DAMAGE();
    PotionEffectType JUMP_BOOST();
    PotionEffectType NAUSEA();
    PotionEffectType RESISTANCE();

    /**
     * Gets the supported version string.
     *
     * @return Version identifier (e.g., "1.20.6", "1.21")
     */
    String getSupportedVersion();

    /**
     * Checks if this adapter supports the current server version.
     *
     * @param serverVersion The server's version string
     * @return true if compatible, false otherwise
     */
    boolean isCompatible(String serverVersion);
}

File: compatibility/api/src/main/kotlin/io/github/kdy05/physicalFighters/api/PotionEffectTypeAdapter.kt
package io.github.kdy05.physicalFighters.api

import org.bukkit.potion.PotionEffectType

interface PotionEffectTypeAdapter : VersionedAdapter {

    fun SLOWNESS(): PotionEffectType
    fun HASTE(): PotionEffectType
    fun MINING_FATIGUE(): PotionEffectType
    fun STRENGTH(): PotionEffectType
    fun INSTANT_HEALTH(): PotionEffectType
    fun INSTANT_DAMAGE(): PotionEffectType
    fun JUMP_BOOST(): PotionEffectType
    fun NAUSEA(): PotionEffectType
    fun RESISTANCE(): PotionEffectType
}
